omod/src/test/java/org/openmrs/module/owa/web/controller/OwaRestControllerTest.java

```diff
@@ -0,0 +1,102 @@
+package org.openmrs.module.owa.web.controller;
+
+import java.io.File;
+import java.io.FileInputStream;
+import java.net.URL;
+import java.util.List;
+
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+import org.junit.After;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+import org.mockito.Mockito;
+import org.openmrs.GlobalProperty;
+import org.openmrs.api.context.Context;
+import org.openmrs.api.context.ServiceContext;
+import org.openmrs.messagesource.MessageSourceService;
+import org.openmrs.util.OpenmrsClassLoader;
+import org.openmrs.web.WebConstants;
+import org.openmrs.web.test.BaseModuleWebContextSensitiveTest;
+import org.springframework.mock.web.MockHttpServletRequest;
+import org.springframework.mock.web.MockHttpServletResponse;
+import org.springframework.mock.web.MockMultipartFile;
+import org.springframework.mock.web.MockServletContext;
+
+public class OWARestControllerTest extends BaseModuleWebContextSensitiveTest{
+	
+	MessageSourceService originalMessageSourceService;
+	
+	MessageSourceService messageSourceService;
+	
+	public OWARestControllerTest() {
+		
+	}
+	
+	@Before
+	public void setUpMockMessageSourceService() {
+		originalMessageSourceService = ServiceContext.getInstance().getMessageSourceService();
+		messageSourceService = Mockito.mock(MessageSourceService.class);
+	}
+	
+	@After
+	public void restoreOriginalMessageSourceService() {
+		if (originalMessageSourceService != null) {
+			ServiceContext.getInstance().setMessageSourceService(originalMessageSourceService);
+		}
+	}
+	
+	@Test
+	public void getAppList_shouldNotReturnNull() throws Exception {
+		OwaRestController controller = (OwaRestController) applicationContext.getBean("owaRestController");
+		Assert.assertNotNull(controller.getAppList());
+	}
+	
+	@Test
+	public void getSettings_shouldNotReturnNull() throws Exception {
+		OwaRestController controller = (OwaRestController) applicationContext.getBean("owaRestController");
+		Assert.assertNotNull(controller.getSettings());
+	}
+	
+	@Test
+	public void updateSettings_shouldNotReturnNull() throws Exception {
+		OwaRestController controller = (OwaRestController) applicationContext.getBean("owaRestController");
+		List<GlobalProperty> settings = Context.getAdministrationService().getGlobalProperties();
+		Assert.assertNotNull(controller.updateSettings(settings));
+	}
+	
+	@Test
+	public void upload_caseNotAZipFile() throws Exception {
+		HttpServletRequest request = new MockHttpServletRequest(new MockServletContext(), "POST", "/module/owa/owaRest.htm");
+		HttpServletResponse response = new MockHttpServletResponse();
+		URL url = OpenmrsClassLoader.getInstance().getResource("testing");

```

Comment:
In above line url will be null. So replace that with FileInputStream file = new FileInputStream(new File("src/test/resources/testing")); and use this FileInputStream in below line instead of "new FileInputStream(url.getFile())".